### PR TITLE
blockdev_mirror_readonly: block mirror on readonly device

### PR DESCRIPTION
**Pull request by zhencliu**
Signed-off-by: Zhenchao Liu <zhencliu@redhat.com>

ID: 1891436

### CODE REVIEW

**Comment by zhencliu** [comment]
local fs:
 (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_mirror_readonly.q35: PASS (80.38 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-26T22.10-9d5180f/results.html
JOB TIME   : 82.92 s

iscsi direct:
 (1/1) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_mirror_readonly.q35: PASS (82.43 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-26T22.19-449a7de/results.html
JOB TIME   : 85.38 s

ceph:
 (1/1) Host_RHEL.m8.u3.product_av.ceph.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_mirror_readonly.q35: PASS (73.28 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-26T23.53-6cf4bd9/results.html
JOB TIME   : 76.12 s

nbd:
 (1/1) Host_RHEL.m8.u3.product_av.nbd.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_mirror_readonly: PASS (72.72 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-27T02.45-4eb9654/results.html
JOB TIME   : 75.48 s

gluster direct:
 (1/1) Host_RHEL.m8.u3.product_av.gluster_direct.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_mirror_readonly.q35: PASS (112.63 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-27T14.58-c0f6e06/results.html
JOB TIME   : 117.62 s

hi @YongxueHong @ngu-niny , can you please help to review this patch? Thanks.

File: qemu/tests/blockdev_mirror_readonly.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        self.preprocess_data_disks()
+        self.prepare_main_vm()
+        self.add_target_data_disks()
+
+    def check_mirrored_block_nodes_readonly(self):
+        for tag in self.params.objects("source_images"):
+            try:
+                self.format_data_disk(tag)
+            except ShellCmdError as e:
+                if not bool(re.search(self.params['error_msg'], str(e), re.M)):
```
Comment (YongxueHong) [inline qemu/tests/blockdev_mirror_readonly.py:23]:
```suggestion
                if not re.search(self.params['error_msg'], str(e), re.M):
```
```
not re.search(r"s", "sss")
False
not re.search(r"x", "sss")
True
```

Comment (zhencliu) [reply qemu/tests/blockdev_mirror_readonly.py]: updated, thx.

**Comment by zhencliu** [comment]
(1/1) Host_RHEL.m8.u3.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.blockdev_mirror_readonly.q35: PASS (80.35 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kar/workspace/job-results/job-2020-10-27T05.35-918c599/results.html
JOB TIME   : 82.96 s

**Review by YongxueHong — APPROVED** [review]

**Comment by ngu-niny** [comment]
LGTM and test PASS:

(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_readonly: PASS (108.50 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**